CSharp/JobPrep/Program.cs

```diff
@@ -0,0 +1,333 @@
+            System.Net.ServicePointManager.DefaultConnectionLimit = 20;
+
+            if (File.Exists("..\\..\\batchcred.txt"))
+            {
+                string[] lines = File.ReadAllLines("..\\..\\batchcred.txt");
+                if (lines.Length >= 2)
+                {
+                    BatchAccount = lines[0];
+                    BatchKey = lines[1];
+                }
+            }
+
+            // Get an instance of the BatchClient for a given Azure Batch account.
+            BatchCredentials cred = new BatchCredentials(BatchAccount, BatchKey);
+            using (IBatchClient client = BatchClient.Connect(Url, cred))
+            {
+                // if you want to put a retry policy in place, enable it here
+                // the built-in policies are No Retry (default), Linear Retry, and Exponential Retry
+                //client.CustomBehaviors.Add(new SetRetryPolicy(new Microsoft.Azure.Batch.Protocol.LinearRetry()));
+
+                ListPools(client);
+                ListWorkItems(client);
+
+                CreatePoolIfNotExist(client, Program.PoolName);
+                AddWork(client);
+
+                ListPools(client);
+                ListWorkItems(client);
+            }
+
+            Console.WriteLine("Press return to exit...");
+            Console.ReadLine();
+        }
+
+        private static void CreatePoolIfNotExist(IBatchClient client, string poolName)
+        {
+            // All Pool and VM operation starts from PoolManager
+            using (IPoolManager pm = client.OpenPoolManager())
+            {
+                // go through all the pools and see if it already exists
+                bool found = false;
+                foreach (ICloudPool p in pm.ListPools())
+                {
+                    // pools are uniquely identified by their name
+                    if (string.Equals(p.Name, poolName))
+                    {
+                        Console.WriteLine("Using existing pool {0}", poolName);
+                        found = true;
+
+                        if (!p.ListVMs().Any<IVM>())
+                        {
+                            Console.WriteLine("There are no VMs in this pool. No tasks will be run until at least one VM has been added via resizing.");
+                            Console.WriteLine("Resizing pool to add 3 VMs. This might take a while...");
+                            p.Resize(3);
+                        }
+
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    Console.WriteLine("Creating pool: {0}", poolName);
+                    // if pool not found, call CreatePool
+                    //You can learn more about os families and versions at:
+                    //http://msdn.microsoft.com/en-us/library/azure/ee924680.aspx
+                    ICloudPool pool = pm.CreatePool(poolName, targetDedicated: 20, vmSize: "small", osFamily: "3");
```

Comment:
Creating a pool with 20 VMs seems excessive.  Does this sample require that many VMs?  On line 90 above, you're only adding 3 VMs if none are found in the pool.